src/bp/core/repositories/telemetry_payload.ts

```diff
@@ -0,0 +1,53 @@
+import { DataRetentionService } from 'core/services/retention/service'
+import { inject, injectable } from 'inversify'
+import Knex from 'knex'
+import _ from 'lodash'
+import ms from 'ms'
+
+import Database from '../database'
+import { TYPES } from '../types'
+
+export interface TelemetryPayloadRepository {
+  getPayload(uuid: string): Promise<any>
+  insertPayload(uuid: string, url: string, payload: JSON): Promise<void>
+  removePayload(uuid: string): Promise<any>
```

Comment:
```suggestion
  removePayload(uuid: string): Promise<TelemetryPayload>
```
And you should define a Typescript type for `TelemetryPayload`